doc/performance-investigations/0011-gunicorn-worker-thread-config/summary.md

```diff
@@ -0,0 +1,100 @@
+| 7 x 2           | 68              | 120             | 150             | 210             | 300               | 730              | 199                     | 214                      | 206                             | 325,552        |
+| 9 x 2           | 65              | **110**         | **130**         | **190**         | **270**           | 700              | 168                     | **190**                  | 179                             | 328,011        |
+| 9 x 6           | 63              | **110**         | **130**         | **190**         | 280               | 560              | 160                     | 196                      | 178                             | 328,588        |
+| 9 x 9           | 68              | 120             | 150             | 200             | 300               | 740              | 172                     | 213                      | 192                             | 326,285        |
+| 9 x 10          | 67              | 120             | 140             | 200             | 300               | 580              | 172                     | 214                      | 193                             | 326,015        |
+| 10 x 7          | 65              | 120             | 140             | 200             | 310               | 660              | 172                     | 212                      | 192                             | 326,886        |
+| 10 x 10         | 73              | 130             | 160             | 230             | 340               | 620              | 196                     | 246                      | 221                             | 323,284        |
+
+- There were 0 failures for these configurations.
+
+### Performance comparison of thread counts
+
+The following results were omitted because the 99th percentile response was above 500ms:
+1. All workers with a thread count of 1
+2. All permutations of 1 and 2 workers.
+
+#### Workers: 3
+
+![](images/performance_graph_workers3.png)
+
+#### Workers: 4
+
+![](images/performance_graph_workers4.png)
+
+#### Workers: 5
+
+![](images/performance_graph_workers5.png)
+
+#### Workers: 6
+
+![](images/performance_graph_workers6.png)
+
+#### Workers: 7
+
+![](images/performance_graph_workers7.png)
+
+#### Workers: 8
+
+![](images/performance_graph_workers8.png)
+
+#### Workers: 9
+
+![](images/performance_graph_workers9.png)
+
+#### Workers: 10
+
+![](images/performance_graph_workers10.png)
+
+#### Comparison of the 99th percentile between worker and thread counts
+
+![](images/performance_graph_workers_by_threads.png)
+
+## Observations
+
+- Using a thread count of 1 or worker count of 1/2 is not feasible.
+- Our current configuration of 7 workers by 7 threads is not the most optimal.
+
+## Decision
+- Since different configuration has different pros and cons for different percentiles a few different worker/thread configurations may need further testing.
```

Comment:
I think we should be explicit and state here which worker/thread configs should be tested

Reply:
Yup, waiting for us to make a decision first.